### PR TITLE
M2 #8 testing library

### PR DESCRIPTION
**Pull request by Allandrow**
Adds a series of unit and integration tests to verify the logic of adding, removing and filtering decks.

Unit tests covers the recoil selectors dependencies between the different atoms and the correct filtering of values based on those dependencies.
Integration tests render the `Listing` component and simulate user interaction on DOM nodes to verify that actions are correctly called and observe the displayed result.

### Notables Changes:

- Formats list is now a simple import from constants instead of fetching it.
- Make exports from vitest globally accessible in test files, allowing the use of exported methods without the need to individually import them each time.

### Breaking Changes:
- Updated the required NodeJS version to `20`. This version implements web apis (like `crypto`, used to generate uuids for decks) and allows us to bypass stubbing those apis.

### CODE REVIEW

**Comment by Allandrow** [comment]
The `Listing.test.tsx` file might need some slight refactorization still, there is a quite a lot of duplication in the different test blocks that could be improved

File: src/modules/listing/FormatFilter.tsx [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
+import { formatList } from '@/config';
+import { Format } from '@/types';
 
 export default function FormatFilter() {
   const [formatFilter, setFormatFilterState] =
     useRecoilState(formatFilterState);
 
-  const { data: formats } = useSWR<Format[]>(endpoints.formats, fetcher);
-
   const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
```
Comment (Kazay) [inline src/modules/listing/FormatFilter.tsx:11]: I know it's mine but we could use an import to keep writing consistent across app.

Comment (Allandrow) [reply src/modules/listing/FormatFilter.tsx]: Updated, now the `ChangeEvent` type is imported

File: src/store/withFilters.test.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import { decks } from '@/test';
+
+/**
+ * Selector tests are based on recoil snapshots where you set the values of atoms and observe
+ * the result via the snapshot.getLoadable method
+ * More info: https://recoiljs.org/docs/guides/testing#testing-recoil-state-outside-of-react
+ */
+
+describe('withFilters selectors', () => {
+  describe('Search term only', () => {
```
Comment (Kazay) [inline src/store/withFilters.test.ts:18]: I'd match terms used when declaring atoms : 

`Name filter only`

**Review by Kazay — APPROVED** [review]

File: src/store/withFilters.test.ts (hunk 2 above)
Comment (Allandrow) [reply src/store/withFilters.test.ts]: Mentions of the word "search" have been changed across the different test files to match this convention

**Pull request merged**